spec/jobs/cleanup_job_spec.rb

```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe CleanupJob, type: :job do
+  let!(:user) { create(:user) }
+  let!(:targets) { create_list(:target, 2, user:) }
+  let!(:old_target) { create(:target, user:, created_at: 2.weeks.ago) }
+
+  it 'deletes targets older than one week' do
```

Comment:
You can also validate that the target that is actually being deleted is the `old_target` and not the others. Something like

expect(Target.all).to include(active_target)

```

And something similar to check that the active targets are still alive.
it 'only delete old targets' do
   subject
   expect(Target.all).not_to include(old_target)
end
```

Reply:
Done!